adrs/4031218897/README.md

```diff
@@ -0,0 +1,50 @@
+# React Hook Arguments Convention
+
+- **State:** Draft
+- **Created:** 2022-12-30
+- **Tags:** react, react-hooks
+
+## Context
+
+### Positioned Arguments vs. Single Argument
+
+Positioned arguments, meaning `function (arg1, arg2, arg3)`, suffer from the
+problem that they don’t have names when you use them as a caller, working only
+for simple situations with one or maybe two arguments.
+
+Having one or two arguments is rarely the case, and even that, needs
+clarification. For example, given the following hook
+calling `useFetchPerson(arg1, arg2)`, you can not tell what is what by just
+looking at the shape vs. `useFetchPerson({firstName: "", lastName: ""})` having
+an object brings more clarity because the keys have names.
+
+Objects allow you to avoid breaking changes since you can always add new keys
+without removing or renaming the old ones in the worst case and without having
+to refactor so much all over the place.
+
+When you are inside the hook (unless you have a destructuring assignment due to
+default values), having a single variable that lets you know what values are
+static is beneficial since the cognitive load to figure out what are computed
+values and what is pass-thru is lower if you see `props.[name of something]` you
+can assume it is a static value, which is passed to your hook. Knowing that you
+will need to look up outside the hook if you have an issue or want to see where
+the value comes from is straightforward.
```

Comment:
If I summarized this, I would say:

Using a single object argument is 1) safer for refactors and 2) safer for contract changes.


Example: 


If you need to drop an argument with positional arguments.

```ts
// before
hook(string, string, string)
// after
hook(string, string)
```

It's very easy to make a mistake here doing this work.

--

IN CONTRAST, using an object argument:

```ts
// before
hook({first: string, middle:string, last:string })
// after
hook({first: string, last:string })
```

It would be very difficult to mess that change up.


In code review, you'd see something like this

```ts
// whoops
hook({first: props.first, last: props.middle })
```

And it sticks out like a sore thumb.